src/main/C594C.c

```diff
@@ -108,17 +120,100 @@ INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D6700);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D6780);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D67A0);
+void func_800D67A0(struct QuadObj* arg0)
+{
+    D_8010FCAC[arg0->state](arg0);
+}
 
-INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D67DC);
+void func_800D67DC(struct QuadObj* arg0)
+{
+    struct PlayerObj* ptr = &D_801418C8;
+    register struct Unk* temp_a2 asm("a2") = arg0->unk5C; // fake
```

Comment:
Hang on, I think I have a fix for this...

Reply:
It sort of looks like it should have arguments but I tried and wasn't able to get a match